model/llm/prompt/parse_test.go

```diff
@@ -38,13 +38,26 @@ func TestParseResponse(t *testing.T) {
 		}
 	`)
 
+	validate(t, &testCase{
+		Name: "Empty Response",
+
+		ExpectedAssessment: metrics.Assessments{
+			metrics.AssessmentKeyResponseNotEmpty: 0,
+			metrics.AssessmentKeyResponseNoExcess: 0,
+			metrics.AssessmentKeyResponseWithCode: 0,
+		},
+		ExpectedCode: "",
+	})
+
 	validate(t, &testCase{
 		Name: "Only Code",
 
 		Response: code,
 
 		ExpectedAssessment: metrics.Assessments{
-			metrics.AssessmentKeyResponseNoExcess: 1,
+			metrics.AssessmentKeyResponseNotEmpty: 1,
+			metrics.AssessmentKeyResponseNoExcess: 0,
```

Comment:
Since it is only code, shouldn't `AssessmentKeyResponseNoExcess:` be set to 1 then?

Reply:
Yes. But it does not have a code fence. So we cannot decide if its code or not.

That's what the comment `// If we cannot distinguish between code and text, we sadly also cannot check if the response contains actual code or if there is any excess response content.` says.

Then we are back at this https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1568814274 problem.

I'm trying to correct for such cases with https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1570227179.

Reply:
I see. Can you comment the test case as well please.

The correction of https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1570227179 does not make sense to me. Either the function that analyses the response says it is code or not. Should not be determined afterwards. Setting that metric later, is a behavioral bug to me.

Reply:
Alrightsy. Will comment and remove the hacky correction afterwards. Until we have some way of detecting if it the response resembles source code. We will have scoring issues then (where the execution runs fine but we don't detect that it's source code).

Reply:
Does that mean, we cannot replicate the manual-log-checking-results? Or what is the problem? Don't get it.

Reply:
The problem is when the model does **not** respond with code fences. We currently have no way of telling within `GenerateTestsForFile` if the response is code or just text (https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1568814274).

In the manual log checking we were able to decide if a model was trying to produce code or just text easily by just looking at it.

Because of goodwill, we currently assume that a response without code fences is code and forward it to the execution (cause that will of course fail if it's not code). I was trying to correct for these errors in hindsight in a hacky way: https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1570227179 (basically awarding points in hindsight after the execution ran successfully).

Reply:
> So to be as good as manual categorization. We must implement [#39 (comment)](https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1568814274) right away.

Is that the only problem we have? Then let's just document the heck out of it in the implementation+test and move on. If we can do every result automatically except for those few, we can check what that means and if there are too many lost, then we iterate.